### PR TITLE
Unfucks SMES construction/deconstruction

### PR DESCRIPTION
**Pull request by Core0verload**
This PR changes SMES construction/deconstruction to be more intuitive.

* Adding a power terminal to SMES updates its status, forcing it to start working. You no longer need to wrench the SMES to activate it.
* Removing SMES terminal is now a required step in SMES deconstruction. No more orphaned terminals left behind.
* SMES that don't work because they lack a terminal now have an examine line.
* Orphaned terminals can now be deconstructed.

### CODE REVIEW

**Comment by Razharas** [comment]
Maybe its time to move away from terminals?
SMES can work on cables alone and nother else in the world uses them

**Comment by Core0verload** [comment]
APCs use them, and, unlike APCs, SMES _can't_ work on cables alone. They have separate input and output.

**Comment by Razharas** [comment]
I personally put a code for SMESes to be able to work on cables instead of terminals
Unless someone removed it it should work

**Comment by Core0verload** [comment]
looks like bullshit, probably is

**Comment by Shadowlight213** [comment]
There literally is no way for them to work without terminals.
They need SEPERATE inputs and outputs.
They use cables for output. How the fuck would you have a seperate cable for input?

File: code/modules/power/smes.dm [hunk 1]
```diff
@@ -41,6 +41,10 @@
 	var/static/list/smesImageCache
 
 
+/obj/machinery/power/smes/examine(user)
+	..()
+	user << "<span class='warning'>This SMES has no power terminal!</span>"
```
Comment (optimumtact) [inline code/modules/power/smes.dm]: shouldn't you check terminal or some other var ?

File: code/modules/power/smes.dm [hunk 2]
```diff
@@ -174,6 +178,15 @@
 		log_game("[src] has been deconstructed by [key_name(user)]")
 		investigate_log("SMES deconstructed by [key_name(user)]","singulo")
 		return
+	else if(istype(I, /obj/item/weapon/crowbar))
```
Comment (optimumtact) [inline code/modules/power/smes.dm]: why are you changing the crowbar to do nothing at all to  SMES?

File: code/modules/power/smes.dm [hunk 3]
```diff
@@ -174,6 +178,15 @@
...
+		return
+
+	return ..()
+
+/obj/machinery/power/smes/default_deconstruction_crowbar(obj/item/weapon/crowbar/C)
+	if(istype(C) && terminal)
+		usr << "<span class='warning'>You must first remove the power terminal!</span>"
+		return 0
```
Comment (optimumtact) [inline code/modules/power/smes.dm]: use the define

**Pull request merged**